src/bin/common/solving_loop.ml

```diff
@@ -514,6 +514,14 @@ let main () =
             (* TODO: add the location of the statement. *)
             Printer.print_smtlib_err "No model produced.";
             st
+        else if Stdlib.(Options.get_sat_solver () <> Tableaux) then
+          begin
+            (* TODO: add the location of the statement. *)
+            Printer.print_smtlib_err
+              "get-model is not supported with the SAT solver %a."
```

Comment:
Can we actually reach this point? I think that if the SAT solver is not Tableaux, we will already fail when `set-option :produce-models` is encountered, and if `set-option :produce-models` is not encountered, we will fail as well.

(Actually, do we? We probably should just error out with "model generation is disabled" here, independenty of the solver, no?)

Reply:
Yes, we can reach this point. This is the very purpose of this PR to add an appropriate error message in this case. 
This error occurs when we call Alt-Ergo without specifying the sat solver but using `get-model` in the input file. Another solution would be to change automatically the SAT solver to Tableaux when we see a `set-option :produce-models`.

Reply:
We cannot do that because `set-option :produce-models` happens after we have chosen the SAT solver.

But what I don't understand is if we call Alt-Ergo without a solver but use `get-model` *without setting `--interpretation`* we should have an error that the interpretation is `INone`, so I don't understand why we proceed past that part. Maybe we don't have that error.

To clarify:

 - If we see a `get-model` and the `interpretation` is `INone`, we must error out because we are not in model-generation mode. This is independent of the solver used. I am not sure we actually do this currently but we must do it.
 - To reach `get-model` with a non-`INone` interpretation, we must either have a `(set-option :produce-models)`, or set the `--interpretation` or `--produce-model` command line flags.
 - `(set-option :produce-models)` should error out if the solver is not Tableaux (I think it does, but am not actually sure)
 - The flags `--interpretation` and `--produce-model` should force the solver to Tableaux, or error out if another solver is explicitly provided

Reply:
Oh yes, you are right. I tried to do it last month, I forgot this issue. 

- Currently, if we use `(set-option :produce-models)`, AE actives the generation of models but you have to choose the SAT-solver. 
https://github.com/OCamlPro/alt-ergo/blob/88ea46122d618cb801379dd47e510f379b36dda8/src/bin/common/solving_loop.ml#L354-L363
It doesn't produce an error if you use the flag `--interpretation none`. 
If you reach a `get-model` without using `set-option :produce-models` or the flags `--interpretation`, `--produce-model`, `--dump-models`, AE displays a smtlib error.
- It is the case. But if we reach 
- `(set-option :produce-models)` produces an error if we don't choose an appropriate SAT solver.
- I confirm that AE prints an error if we try to use another SAT solver with the flags `--produce-model` and `--interpretation` 
forces the SAT-solver to Tableaux.

Reply:
Thanks for confirming! Then it seems to me that this new error is redundant with the existing one below (which should mention `--produce-models` in addition to `:produce-models`, I guess), no?

Reply:
But we don't fail with the first message. The first message tells us you have to select an appropriate SAT solver.  It's just a warning. The second one is a smtlib error since you cannot produce the requested model. We shouldn't ignore silently the statement `get-model`.

Reply:
Ah, then that's the actual issue. We should fail if we see `(set-option :produce-models true)` but can't produce models due to a wrong solver being selected.

Reply:
Err, I mean, we should just ignore the command. But then we fail because we can't produce a model, right?

Reply:
Actually after re-reading the smtlib spec I believe we should "fail" in the sense of returning an `smtlib` error but then continue as if the `set-option` was not encountered.

I think this PR should also update the message when we see a `get-model` but we are not in model-production mode; currently it says to use `(set-option :produce-models true)` but it should say to use `--produce-models` instead, because otherwise there will immediately be another error due to Tableaux not being selected.

(And maybe add tests for the combinations of `--produce-models` / `set-option` / `get-model`?)

Reply:
Ok, we can add some cram tests but we should fix the issue with Mac OS first. 

I will print a smtlib error but we don't print anything if there is `(get-model)` after a failed `set-option`?

Reply:
We do I think? It's the message that currently says "please use set-option :produce-models"